src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,121 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] items;
+    private int size;
+
+    public ArrayList() {
+        this.items = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
+        if (size == items.length) {
+            items = increaseCapacity();
+        }
 
+        items[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+
+        if (size == items.length) {
+            items = increaseCapacity();
+        }
+
+        System.arraycopy(items, index, items, index + 1, size - index);
+
+        items[index] = value;
 
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
```

Comment:
Consider adding a null check for the `list` parameter to prevent a `NullPointerException` when `addAll` is called with a null argument.